File: backend/app/api/routes/expenses.py
"""CRUD de despesas do petshop."""

from fastapi import APIRouter, HTTPException, status
from sqlalchemy import select

from app.api.deps import CurrentUser, DbSession
from app.models.expense import Expense
from app.schemas.expense import ExpenseCreate, ExpenseOut

router = APIRouter(prefix="/expenses", tags=["expenses"])


@router.get("", response_model=list[ExpenseOut])
def list_expenses(user: CurrentUser, db: DbSession) -> list[Expense]:
    return list(db.scalars(select(Expense).order_by(Expense.expense_date.desc())))


@router.post("", response_model=ExpenseOut, status_code=status.HTTP_201_CREATED)
def create_expense(payload: ExpenseCreate, user: CurrentUser, db: DbSession) -> Expense:
    expense = Expense(**payload.model_dump())
    db.add(expense)
    db.commit()
    db.refresh(expense)
    return expense


@router.delete("/{expense_id}", status_code=status.HTTP_204_NO_CONTENT)
def delete_expense(expense_id: int, user: CurrentUser, db: DbSession) -> None:
    expense = db.get(Expense, expense_id)
    if expense is None:
        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="Despesa não encontrada.")
    db.delete(expense)
    db.commit()

File: src/api/expenses.ts
import { apiRequest } from './client'
import type { Expense } from '../types'

export interface ExpenseInput {
  description: string
  amount: number
  expense_date: string
}

export function listExpenses(): Promise<Expense[]> {
  return apiRequest<Expense[]>('/expenses')
}

export function createExpense(input: ExpenseInput): Promise<Expense> {
  return apiRequest<Expense>('/expenses', { method: 'POST', body: input })
}

export function deleteExpense(id: number): Promise<void> {
  return apiRequest<void>(`/expenses/${id}`, { method: 'DELETE' })
}
